### PR TITLE
feat(motion_velocity_smoother): add a diagnostic for smoothing failure

### PR DESCRIPTION
**Pull request by mkuri**
## Description

Fixed a problem with slow deceleration when rapid deceleration is required

## Related links

https://tier4.atlassian.net/browse/RT0-24965

## Tests performed

1. Spawn obstacles in front of the vehicle to rapid deceleration
2. The diagnostics of "smoother_errors" is triggered and emergency stop is performed.

Screenshot from 2023-06-14 23-10-31
Screenshot from 2023-06-14 23-11-44

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by 0x126** [comment]
@mkuri `enable_diagnostics_when_smoothing_failure`がdoubleで初期値が0.0になってますが、boolにした方がいいと思います。

File: planning/motion_velocity_smoother/config/default_motion_velocity_smoother.param.yaml [hunk 1]
```diff
@@ -50,3 +50,6 @@
 
     # system
     over_stop_velocity_warn_thr: 1.389  # used to check if the optimization exceeds the input velocity on the stop point
+
+    # diagnostics
+    enable_diagnostics_when_smoothing_failure: 0.0
```
[the remark this answers is not in the document]
Comment (mkuri) [reply planning/motion_velocity_smoother/config/default_motion_velocity_smoother.param.yaml:55]: boolにしたほうが良いんですが、smoother nodeのparam updateの関数がdoubleにしか対応していなくて結構面倒だったのでdoubleにしました（動的なparam変更に対応しないのはありですね…）

File: planning/motion_velocity_smoother/src/motion_velocity_smoother_node.cpp [hunk 2]
```diff
@@ -579,6 +620,10 @@ bool MotionVelocitySmootherNode::smoothVelocity(
   if (!smoother_->apply(
         initial_motion.vel, initial_motion.acc, clipped, traj_smoothed, debug_trajectories)) {
     RCLCPP_WARN(get_logger(), "Fail to solve optimization.");
+    if (node_param_.enable_diagnostics_when_smoothing_failure) {
+      failure_to_apply_smoother_ = 1.0;
```
[the remark this answers is not in the document]
Comment (mkuri) [reply planning/motion_velocity_smoother/src/motion_velocity_smoother_node.cpp:624]: ごっちゃになってました。trueですね。c++の曖昧な仕様に助けられてました

File: planning/motion_velocity_smoother/config/default_motion_velocity_smoother.param.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkuri) [reply planning/motion_velocity_smoother/config/default_motion_velocity_smoother.param.yaml]: https://github.com/tier4/autoware.universe/pull/578/commits/29268aed43de85a49c617dc739eadd93b55fcae8

File: planning/motion_velocity_smoother/src/motion_velocity_smoother_node.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkuri) [reply planning/motion_velocity_smoother/src/motion_velocity_smoother_node.cpp]: https://github.com/tier4/autoware.universe/pull/578/commits/4386d6ab6c71b9e26e5a4644939679837cc4665f

**Comment by mkuri** [comment]
修正後のソフトで動作確認ok
Screenshot from 2023-06-14 23-23-12

File: planning/motion_velocity_smoother/config/default_motion_velocity_smoother.param.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkuri) [reply planning/motion_velocity_smoother/config/default_motion_velocity_smoother.param.yaml]: launcher: https://github.com/tier4/autoware_launch.x2/pull/305/commits/56fa40a89eea0ee85b72ca646f87cec593789c9f

**Review by 0x126 — APPROVED** [review]

**Comment by TomohitoAndo** [comment]
pre-commit.ci run

**Pull request closed without merging**